blackbird_python/blackbird/listener.py

```diff
@@ -427,6 +446,28 @@ def exitStatement(self, ctx: blackbirdParser.StatementContext):
         else:
             self._program._operations.append(operation)
 
+    def enterForloop(self, ctx: blackbirdParser.ForloopContext):
+        self._infor = True
+
+    def exitForloop(self, ctx: blackbirdParser.ForloopContext):
+        self._infor = False
+
+        if ctx.INT():
+            _FORVAR = range(int(ctx.INT().getText()))
+        elif ctx.vallist():
+            _FORVAR = ctx.vallist().getText().split(',')
+
+        for var in _FORVAR:
+            if ctx.NAME():
+                try:
+                    var = PYTHON_TYPES[ctx.vartype().getText()](var)
+                except ValueError:
+                    raise ValueError(f"invalid value {var}; must be {ctx.vartype().getText()}")
+                _VAR[ctx.NAME().getText()] = var
```

Comment:
Could this be replaced with the function for evaluating an expression? That way you could have

```python
for [x**2, 54, 1] -> m
```

and it should still work.

I think the function for evaluating expressions also includes the logic for getting variable values.

Reply:
Done! Expressions are now supported. :slightly_smiling_face: